### DISCUSSION TITLE
rullerende placeholdertekst

### DISCUSSION THREAD

**Discussion by kristineSorensen** [fremtind/jokul#2541] [Annet]
På forsiden til meldeskade har vi eksperimentert med et søkefelt som gir svar i chatboten. Målet med det er å hjelpe folk til å finne riktig skjema i tillegg til en hypotese om at måten man skriver i et søkefelt genererer bedre resultater i chatbot. https://meldskade.fremtind.no/#/?company=sb1 

Vi ser nå på placeholderteksten og har fått inn en rekke eksempeltekster basert på det kunder hyppigst spør chatboten om. Det er et ønske om å ha rullerende placeholdertekst med forskjellige eksempler. Det vil gi muligheten til å vise flere eksempler på hva man kan søke på og det vil skape en bevegelse som om noen skriver noe.
Eksempel: https://codepen.io/joelewis/pen/ePOrmV 

Vi sitter på flere spørsmål som jeg tenker å ta opp i forum, men jeg starter en diskusjon her i tillegg. 
- må noe som dette inn i komponenten? 
- Hva med UU? er dette en bevegelse som må kunne slås av? 
- Er det andre som har behov for noe ligende?

**Comment by wkillerud** [comment]
> * må noe som dette inn i komponenten?

Nei, det er teknisk sett allerede mulig i dag: https://codesandbox.io/s/damp-pine-qmw60?file=/src/App.tsx (refresh for å se animasjonen på nytt)

> * Hva med UU? er dette en bevegelse som må kunne slås av?

Jeg mener ja. Introteksten på https://jokul.fremtind.no respekterer [`prefers-reduced-motion`-innstillingen](https://github.com/fremtind/jokul/blob/main/portal/src/components/Delaytext/DelayText.tsx) fra nettleseren. Introteksten er så klart mer fremtredende, men tenker det er greit å gjøre det samme i dette tilfellet.

---

Det sagt, jeg kommer ikke på å ha sett noe sånt som dette så mange plasser. Personlig [^1] er det bevegelse et sted jeg ikke forventer å se det. Det er kanskje et poeng å tiltrekke seg oppmerksomhet og sånn sett "working as intended". På en anne side, om dette er tekst som er viktig at brukeren får med seg bør man vurdere en hjelpetekst under søkefeltet i stedet.

Eventuelt tenke utenfor (den bokstavelige) boksen 😄 CSS-Tricks har en kul variant (riktignok i modal) hvor et klikk på søkefeltet viser en rekke populære søketreff. Google har en [autosuggest](https://jokul.fremtind.no/komponenter/autosuggest)-ish sak, med ulempen at man må ha _noe_ i søkefeltet for å få forslag.

![Screenshot of CSS Tricks search UI](https://user-images.githubusercontent.com/1223410/143466156-24269f3e-8dba-48bd-bf40-cb49a1bc0aa6.png)


[^1]:  Nei, jeg er ikke brukeren 😄

**Reply by kristineSorensen** [reply]
Takk for innspill :) 

https://codepen.io/sroucheray/pen/qOdZwJ 

Her er en annen variant som Ivana tipset meg om. En "rullerende" bevegelse der man får sett hele setningen med en gang. 
Finn.no har også placholdertekster som varierer, men de er statiske og byttes ved at man refresher siden.

**Comment by piofinn** [comment]
Hvor i chatbot'en er dette tenkt å brukes? Hvis det ser ut som det blir skrevet noe i feltet man bruker for å skrive til chatbot-en må man være veldig sikker på at ikke det oppleves som utrygt. Merker at jeg generelt er litt skeptisk, og kanskje tenker at man kan vurdere en løsning á la finn.no? **Edit**: Skjønte nå at det var et søkefelt under overskriften på siden 😅 Tenker det at jeg ikke så det kanskje understøtter poenget under..

Hvis brukerne ikke _finner_ søkefeltet enkelt nok tenker jeg at man bør vurdere å endre design/layout om mulig, heller enn å legge inn bevegelse for å "lokke til seg" brukeren.

> om dette er tekst som er viktig at brukeren får med seg bør man vurdere en hjelpetekst under søkefeltet i stedet.

Viktig punkt fra @wkillerud 👆 Placeholder-tekst blir _ikke_ lest opp av skjermleser.

**Reply by Slaine2** [reply]
Kanskje jeg mangler målet, men jeg elsker virkelig muligheten til å få forslag mens jeg skriver. Kanskje noe slikt vil være veldig nyttig her?
https://www.stadtluzern.ch/
